src/targets/services/cleanOrphanAccounts.js

```diff
@@ -0,0 +1,81 @@
+const cozyFetch = require('../../lib/services/cozyFetch')
+
+const logMessagePrefix = level =>
+  `[cozy-collect][cleanOrphanAccounts][${level}]:`
+
+const output = level =>
+  function() {
+    console.log.apply(
+      console,
+      [logMessagePrefix(level)].concat(
+        // Node < 7 does not know about Object.values
+        Object.keys(arguments).map(key => arguments[key])
+      )
+    )
+  }
+
+const log = {
+  error: output('error'),
+  info: output('info')
+}
+
+const handleFetchError = error => log.error(error.message)
+
+log.info('Fetching accounts...')
+
+cozyFetch(
+  'GET',
+  '/data/io.cozy.accounts/_all_docs?include_docs=true',
+  null,
+  true
+)
+  .then(accounts => {
+    log.info(`${accounts.length || 0} account(s) has been found.`)
+    log.info('Fetching konnector triggers...')
+
+    cozyFetch('GET', '/jobs/triggers?Worker=konnector', null, true)
+      .then(result => {
+        const hasTriggers = result && result.data && result.data.length
+        const triggers = result.data
+
+        if (hasTriggers) {
+          log.info(
+            `${triggers.length || 0} konnector trigger(s) has been found.`
+          )
+        }
+
+        const orphanAccounts = hasTriggers
+          ? accounts.filter(
+              account =>
+                !triggers.find(
+                  trigger =>
+                    trigger.attributes &&
+                    trigger.attributes.message &&
+                    trigger.attributes.message.account &&
+                    trigger.attributes.message.account === account._id
+                )
+            )
```

Comment:
Indexing the triggers by `attributes.message.account` would make the intention more clear IMHO.

```
const { indexBy, get } from 'lodash'

const getAccountFromTrigger = trigger => get(t, 'attributes.message.account')
const triggerByAccountId = indexBy(triggers.filter(t => getAccountFromTrigger), getAccountFromTrigger)
const orphanAccounts = accounts.filter(account => triggerByAccountId[accounts._id])
```